.github/workflows/ci-cd.yml

```
@@ -0,0 +1,169 @@
+          # Tests will skip PayPal/Email tests if these aren't set
+          PAYPAL_CLIENT_ID: ${{ secrets.PAYPAL_CLIENT_ID || '' }}
+          PAYPAL_CLIENT_SECRET: ${{ secrets.PAYPAL_CLIENT_SECRET || '' }}
+          EMAIL_PROVIDER: ${{ secrets.EMAIL_PROVIDER || '' }}
+          SKIP_OPTIONAL_SERVICE_TESTS: 'true'  # Allow tests to skip optional service tests
+
+      - name: Generate test coverage
+        run: cd tests && npm run test:coverage
+        continue-on-error: true
+
+      - name: Upload coverage reports
+        uses: codecov/codecov-action@v3
```

actionlint (1.7.9)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)